pkg/controller/servicebindingrequest/binding_test.go

```diff
@@ -148,6 +147,35 @@ func TestServiceBinder_Bind(t *testing.T) {
 					require.True(t, match, "expected action %+v not found", w)
 				}
 			}
+
+			providedBindingPath := args.options.SBR.Spec.ApplicationSelector.BindingPath
+			computedBindingPath := sb.SBR.Spec.ApplicationSelector.BindingPath
+
+			if providedBindingPath != nil {
+
+				// user has provided BindingPath information explicitly
+				require.True(t, providedBindingPath.PodSpecPath != nil ||
```

Comment:
I assume this is related to the injection of `BindingPath` in `BuildServiceBinder`, and perhaps should be moved to a separate function specialized in validating `BindingPath` values.

In this case, `BuildServiceBinder` could return an error (let's say `InvalidBindingPathErr`) and specified in `wantBuildErr`.

The validation function could be properly tested to cover `BindingPath` semantics and the contract would be the error returned by `BuildServiceBinder`.

Reply:
Sure, will do!

Reply:
Did my best here b142835763bd2e3f68b64191a3b8427d7d41ee77 . Let me know!